src/tests/store.test.js

```diff
@@ -0,0 +1,30 @@
+import { loadState, STORE_VERSION, CURRENT_STORE_VERSION_KEY } from '../redux/reducers/store';
+
+describe('loadState', () => {
+  beforeEach(() => {
+    localStorage.clear();
+  });
+
+  it('should return undefined when state is null', () => {
+    expect(loadState()).toEqual(undefined);
+  });
+
+  describe('should return undefined when store versions don\'t match', () => {
+    it('should return undefined when store version in local storage is null', () => {
+      localStorage.setItem('state', 'state');
+      expect(loadState()).toEqual(undefined);
+    });
+
+    it('should return undefined when store versions don\'t match', () => {
+      localStorage.setItem(CURRENT_STORE_VERSION_KEY, STORE_VERSION - 1);
+      localStorage.setItem('state', 'state');
+      expect(loadState()).toEqual(undefined);
+    });
+  });
+
+  it('should return the right state that was saved', () => {
+    localStorage.setItem(CURRENT_STORE_VERSION_KEY, STORE_VERSION);
+    localStorage.setItem('state', JSON.stringify('state'));
+    expect(loadState()).toEqual('state');
```

Comment:
How do you tell that for instance we have version 1 (that we stored) instead of 2?

Reply:
This test uses the same `STORE_VERSION` constant used when retrieving the state, [here](https://github.com/openmrs/openmrs-ocl-client/blob/0bd1f0c1b4154c7eabdf6410a2498d9fb1456249/src/redux/reducers/store.js#L7), so we'd expect the state to be retrieved correctly in this scenario.
The test that checks for a mismatch in versions is on line 18 where we set the version to be one lower than the same constant.

Reply:
Let me put it this way, what is the right state which was saved?

Reply:
I see.
I have updated the state we check for to be a constant to make this more explicit.
I have also updated the test description.
It would be helpful to point out that `loadState` either returns the state or `undefined` and we do not have a scenario where we have more than one state version in local storage.

Is this okay?